### PR TITLE
fix: make account menu width and accounts on account page responsive

### PR DESCRIPTION
**Pull request by rolznz**
### Describe the changes you have made in this PR

- Make the account menu width responsive based on screen size and ensure buttons are not pushed off the screen
- Fix long account name causing a scrollbar in the accounts page

Note: For the account menu this is a bit of a hacky solution because it has to manually account for the menu icon and padding. I couldn't find a better one. Technically this is also an issue in the accounts page, but I did a simpler fix for this one as I think it's less important to have an exact fit. EDIT: it might just be better to fix the account menu width to the same size as the dropdown and remove all those checks 😅 )

### Link this PR to an issue [optional]

Fixes _#1972_

### Type of change

(Remove other not matching type)

- `fix`: Bug fix (non-breaking change which fixes an issue)

### Screenshots of the changes [optional]
small phone
image

md breakpoint
image

xl screen
image

Accounts page mobile:
image

md breakpoint
image

xl
image

_Add screenshots to make your changes easier to understand. You can also add a video here._

### How has this been tested?

_Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration_

### Checklist

- [x] My code follows the style guidelines of this project and performed a self-review of my own code
- [x] New and existing tests pass locally with my changes
- [x] I checked if I need to make corresponding changes to the documentation (and made those changes if needed)

### CODE REVIEW

**Comment by rolznz** [comment]
I think the dropdown menu width should match the account menu width, what do you think? @reneaaron 

image

**Comment by reneaaron** [comment]
Nice! 

> I think the dropdown menu width should match the account menu width, what do you think? @reneaaron

Yes. 👍

Can we maybe add a little spacing between the menu icon and the account menu? 

image

`mr-4` should be fine 👌

**Comment by rolznz** [comment]
> Nice!
> 
> > I think the dropdown menu width should match the account menu width, what do you think? @reneaaron
> 
> Yes. 👍
> 
> Can we maybe add a little spacing between the menu icon and the account menu?
> 
> image
> 
> `mr-4` should be fine 👌

Thanks for testing!

The max-width class wasn't being applied. I've fixed this now so the gap is there.

image


Also updated to make the menu width match:
image

**Review by escapedcat — APPROVED** [review]
ack, works great.
Maybe the select should just have a fixed width? Looks a bit weird if it jumps when switching accounts.

e2e is failing on master as well, not related to this PR

**Comment by rolznz** [comment]
> ack, works great. Maybe the select should just have a fixed width? Looks a bit weird if it jumps when switching accounts.
> 
> e2e is failing on master as well, not related to this PR

I agree it's weird it changes width when you switch accounts. It would be simpler to just have a fixed width.

**Comment by reneaaron** [comment]
@rolznz Do you plan to still change this to a fixed width?

**Comment by rolznz** [comment]
@reneaaron I made the change
image
image

File: src/app/components/Menu/MenuList.tsx [hunk 1]
```diff
@@ -18,11 +20,13 @@ function List({ position = "left", children }: Props) {
...
     >
       <Menu.Items
-        className={`${
+        className={classNames(
           position === "left"
             ? "left-0 origin-top-left"
-            : "right-0 origin-top-right"
-        } absolute z-50 mt-2 py-1 w-56 rounded-md shadow-lg bg-white ring-1 ring-black ring-opacity-5 focus:outline-none dark:bg-surface-12dp dark:border-0`}
+            : "right-0 origin-top-right",
+          fullWidth ? "w-full" : "w-56",
```
Comment (reneaaron) [inline src/app/components/Menu/MenuList.tsx:27]: Why do we need a width here?

**Review by reneaaron — COMMENTED** [review]
Tested, feels a lot better now and involves less complexity👌

File: src/app/components/Menu/MenuList.tsx (hunk 1 above)
Comment (rolznz) [reply src/app/components/Menu/MenuList.tsx:27]: @reneaaron I think to make the menu a consistent width no matter where it is used, and make sure it does not automatically expand if one of the items is wide. I am not sure I understand the question, does this answer it?

**Pull request merged**